docs/DeveloperGuide.adoc

```diff
@@ -153,7 +153,175 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+=== Question feature
+
+The question feature utilises the `QuestionCommandParser` class to parse the user command input into the different
+command types and validates the input. There are two types of questions `OpenEndedQuestion` and `McqQuestion` which
+extends the `Question` class. Questions are then added into the `QuestionBank#questions` observable list. +
+
+The feature comprises of five commands namely,
+
+* <<Feature-Question-Add, `QuestionAddCommand`>> - Adding questions
+* <<Feature-Question-Edit, `QuestionEditCommand`>> - Editing questions
+* <<Feature-Question-Delete, `QuestionDeleteCommand`>> - Deleting questions
+* <<Feature-Question-List, `QuestionListCommand`>> - Listing questions
+* `QuestionSlideshowCommand` - Starting a slideshow (See <<Feature-Slideshow, slideshow feature>>)
+
+The commands when executed, will interface with the methods exposed by the `Model` interface to perform the related operations
+(See <<Design-Logic, logic component>> for the general overview).
+
+_To Add: Class diagram of the interaction between the question parser and commands_
+
+[[Feature-Question-Add]]
+==== Add command
+
+The following is a detailed explanation of the operations `QuestionAddCommand` performs. +
+
+*Step 1*. The `QuestionAddCommand#execute(Model model)` method is executed and it validates type of question defined.
+The type defined will dictate if the resulting question will be a `OpenEndedQuestion` or `McqQuestion` question type.
+
+*Step 2*. The method `Model#addQuestion(Question question)` will then be called to add the question and a success message will
+be generated by the `QuestionAddCommand#generateSuccessMessage(Question question)` method and a new `CommandResult` will be
+returned with the generated success message.
+
```

Comment:
There could be a sequence diagram explaining the control flows for the add/edit/delete/list commands.
Currently, you only have steps in bullet points, but it's unclear which objects are involved and which are active at what times.